workflow/common/util.go

```diff
@@ -151,9 +151,14 @@ func ProcessArgs(tmpl *wfv1.Template, args wfv1.ArgumentsProvider, globalParams,
 
 				cmValue, err := GetConfigMapValue(configMapInformer, namespace, cmName, cmKey)
 				if err != nil {
-					return nil, errors.Errorf(errors.CodeBadRequest, "unable to retrieve inputs.parameters.%s from ConfigMap: %s", inParam.Name, err)
+					if inParam.ValueFrom.Default != nil {
```

Comment:
should we make sure `err` is IsNotFound?
what about test?

Reply:
> Default specifies a value to be used if retrieving the value from the specified source fails
Based on the original purpose from the doc-string, I think we want to have default no matter what specific error we are getting. 

I'll add tests.

Reply:
I'm not sure about this, should a transient error result in my workflow running with unexpected parameters?

Reply:
That makes sense. Added a check for IsNotFound as well as a test.